Python/Product/TestAdapter/TestContainerDiscovererProject.cs

```diff
@@ -328,81 +331,101 @@ class TestContainerDiscovererProject : ITestContainerDiscoverer, IDisposable {
+                    break;
+                case TestFileChangedReason.Removed:
+                    _testFilesUpdateWatcher.RemoveWatch(e.File);
+                    break;
+            }
+            _isRefresh = true;
+            NotifyContainerChanged();
+        }
+
+        private ProjectInfo FindProjectInfo(string file, IVsProject vsProject = null) {
             if (vsProject == null) {
+                // bschnurr todo: this is only looking at opened files
                 var rdt = (IVsRunningDocumentTable)_serviceProvider.GetService(typeof(SVsRunningDocumentTable));
-                vsProject = VsProjectExtensions.PathToProject(e.File, rdt);
+                vsProject = VsProjectExtensions.PathToProject(file, rdt);
             }
 
-            if (vsProject == null)
-                return;
-
-            string projectHome = vsProject.GetProjectHome();
-            if (projectHome != null &&
-                _projectMap.TryGetValue(projectHome, out ProjectInfo projectInfo)) {
-
-                var sources = new List<string>() { e.File };
-
-                switch (e.ChangedReason) {
-                    case TestFileChangedReason.Added:
-                        UpdateSolutionTestContainersAndFileWatchers(sources, projectInfo, isAdd: true);
-                        break;
-                    case TestFileChangedReason.Changed:
-                        //Need to increment version number so Test Explorer notices a change
-                        UpdateSolutionTestContainersAndFileWatchers(sources, projectInfo, isAdd: true);
-                        break;
-                    case TestFileChangedReason.Removed:
-                        UpdateSolutionTestContainersAndFileWatchers(sources, projectInfo, isAdd: false);
-                        break;
-                    case TestFileChangedReason.Renamed:
-                        UpdateSolutionTestContainersAndFileWatchers(new List<string>() { e.OldFile }, projectInfo, isAdd: false);
-                        UpdateSolutionTestContainersAndFileWatchers(sources, projectInfo, isAdd: true);
-                        break;
-                    default:
-                        //In changed case file watcher observed a file changed event
-                        //In this case we just have to fire TestContainerChnaged event
-                        break;
+            if (vsProject != null) {
+                string projectHome = vsProject.GetProjectHome();
+                if (projectHome != null &&
+                    _projectMap.TryGetValue(projectHome, out ProjectInfo foundProjectInfo)) {
+                    return foundProjectInfo;
                 }
-                NotifyContainerChanged();
             }
-            
+
+            //Renamed  old files are no longer in the project so linear search
+            var projectInfo = _projectMap.Values.FirstOrDefault(p => p.TryGetContainer(file, out _));
+            return projectInfo;
+        }
```

Comment:
there was a bug in the previous logic to find the projectInfo when a file was renamed because its no longer in the 
VsProject but its still in our _projectMap of testcontainers.

Also some events send a IVsProject and so do not so moving this complicated logic into its own function